libr/anal/p/anal_mips_gnu.c

```diff
@@ -35,10 +34,8 @@ static int mips_op(RAnal *anal, RAnalOp *op, ut64 addr, const ut8 *b_in, int len
 	op->addr = addr;
 	r_strbuf_init (&op->esil);
 
-	// Reminder: r_mem_copyendian swaps if arg `endian` ==0 ...
-	// When anal->big_endian is "false", as for mipsel architecture, we NEED to swap here for the below analysis to work.
-	r_mem_copyendian ((ut8*)&opcode, b_in, 4, anal->big_endian ? 1 : 0);
-	r_mem_copyendian (b, b_in, 4, anal->big_endian ? 1 : 0);
+	// Be endian aware
+	opcode = anal->big_endian? read_be32(b): read_le32(b);
```

Comment:
missing spaces before parenthesis. and maybe its better to just `(anal->big_endian?read_be32:read_le32)(b);` or just have a helper function to do that in a single place